### PR TITLE
Once again, ;plot extreme

### PR DESCRIPTION
**Pull request by MetalBall887**
Adding a new graph for showing your extremes each contest

### CODE REVIEW

File: tle/cogs/graphs.py [hunk 1]
```diff
@@ -93,22 +93,19 @@ def _filter_solved_submissions(submissions, contests, tags = []):
                 problems.add(problem_key)
     return solved_subs
 
+def get_extremes(user, contest, submissions, problems):
```
Comment (krofna) [inline tle/cogs/graphs.py]: contest argument seems unused

File: tle/cogs/graphs.py [hunk 2]
```diff
@@ -121,14 +118,70 @@ def _running_mean(x, bin_size):
 
     return res
 
+def _plot_extreme(user, rating_changes, statuses, problemsets, bin_size=3, mark='o', label=''):
+    plot_min, plot_max, times = [], [], []
+
+    for rating_change, status, problems in zip(rating_changes, statuses, problemsets):
+        contest = cf_common.cache2.contest_cache.get_contest(rating_change.contestId)
```
Comment (krofna) [inline tle/cogs/graphs.py]: see above. unused

File: tle/cogs/graphs.py [hunk 3]
```diff
@@ -195,6 +248,52 @@ def __init__(self, bot):
         discord_common.set_author_footer(embed, ctx.author)
         await ctx.send(embed=embed, file=discord_file)
 
+    @plot.command(brief='Plot Codeforces lowest rating unsolved/highest rating solved problems on contests graph', usage='[+zoom] handle')
+    async def extreme(self, ctx, *args: str):
+        """Plots Codeforces lowest unsolved problem/highest solved on codeforces graph for the handle provided."""
+        args = list(args)
+        if '+zoom' in args:
```
Comment (krofna) [inline tle/cogs/graphs.py]: do we need zoom in scatterplot? i think it should be zoomed by default. zoom was only added in rating plot to make it more consistent with cf

File: tle/cogs/graphs.py [hunk 4]
```diff
@@ -195,6 +248,52 @@ def __init__(self, bot):
...
+            zoom = True
+            args.remove('+zoom')
+        else:
+            zoom = False
```
Comment (krofna) [inline tle/cogs/graphs.py]: @MetalBall887 i just realized this is dead code lmao

File: tle/util/codeforces_api.py [hunk 5]
```diff
@@ -165,6 +164,10 @@ class RatingChangesUnavailableError(CodeforcesApiError):
     def __init__(self, comment, contest_id):
         super().__init__(comment, f'Rating changes unavailable for contest with ID `{contest_id}`')
 
+class StatusUnavailableError(CodeforcesApiError):
```
Comment (krofna) [inline tle/util/codeforces_api.py]: @MetalBall887 is this ever used?

File: tle/cogs/graphs.py [hunk 6]
```diff
@@ -93,22 +93,19 @@ def _filter_solved_submissions(submissions, contests, tags = []):
                 problems.add(problem_key)
     return solved_subs
 
+def get_extremes(user, submissions, problems):
 
-def _classify_submissions(submissions):
```
Comment (krofna) [inline tle/cogs/graphs.py]: Why were these moved?

File: tle/cogs/graphs.py [hunk 7]
```diff
@@ -121,14 +118,67 @@ def _running_mean(x, bin_size):
 
     return res
 
+def _plot_extreme(user, rating_changes, statuses, problemsets, bin_size=3, mark='o', label=''):
+    plot_min, plot_max, times = [], [], []
+
+    for rating_change, status, problems in zip(rating_changes, statuses, problemsets):
+        if all([problem.rating for problem in problems]):
```
Comment (krofna) [inline tle/cogs/graphs.py]: You dont need [] in all()

File: tle/util/codeforces_api.py (hunk 5 above)
Comment (MetalBall887) [reply tle/util/codeforces_api.py]: uh no
